clippy_lints/src/needless_borrow.rs

```diff
@@ -36,16 +37,12 @@ declare_clippy_lint! {
     "taking a reference that is going to be automatically dereferenced"
 }
 
-#[derive(Default)]
-pub struct NeedlessBorrow {
-    derived_item: Option<LocalDefId>,
-}
-
+pub struct NeedlessBorrow;
 impl_lint_pass!(NeedlessBorrow => [NEEDLESS_BORROW]);
```

Comment:
If the struct doesn't have fields, you can just do 

```suggestion
declare_lint_pass!(NeedlessBorrow => [NEEDLESS_BORROW]);
```